internal/store/store.go

```diff
@@ -662,84 +650,6 @@ func (s *Store) PushCompose(distro distro.Distro, composeID uuid.UUID, bp *bluep
 	return nil
 }
 
-// PushComposeRequest is an alternative to PushCompose which does not assume a pre-defined distro, as such it is better
-// suited for RCM API and possible future API that would respect the fact that we can build any distro and any arch
-func (s *Store) PushComposeRequest(request ComposeRequest) error {
```

Comment:
I wish you said this during the previous refactoring, because my intention was to replace the `PushCompose` with `PushComposeRequest`. Now it feels like we are just replacing code randomly.

Reply:
> I wish you said this during the previous refactoring, [...]

I'm not sure I understand? Which previous refactoring?

I know that this was your intention, but it didn't happen. Honestly, I don't think we should have merged a function that does something very similar to another function without removing this other function.

My intention here was to merge these two functions, so that it becomes easier to split the queue from the store down the line. It was simpler to keep `PushCompose` than `PushComposeRequest` without sacrificing any existing functionality.

> Now it feels like we are just replacing code randomly.

I thought about this for a very long time. It is not random at all. I'm sorry if I didn't communicate my intention clearly enough in this PR. Hope these comments clear it up a bit.

Reply:
> I'm not sure I understand? Which previous refactoring?

https://github.com/osbuild/osbuild-composer/pull/221

> I know that this was your intention, but it didn't happen. Honestly, I don't think we should have merged a function that does something very similar to another function without removing this other function.

This should have been said in #221 and it didn't happen because it wasn't a priority yet. If you consider this an issue with high priority you could just say so during a planning and I would happily work on it.

> My intention here was to merge these two functions, so that it becomes easier to split the queue from the store down the line. It was simpler to keep PushCompose than PushComposeRequest without sacrificing any existing functionality.

This sounds again like something we were discussing but didn't agree on yet. In cases like this we should first create an issue with description.

> I thought about this for a very long time. It is not random at all. I'm sorry if I didn't communicate my intention clearly enough in this PR. Hope these comments clear it up a bit.

As mentioned above, bigger plans like this should start with an issue that describes your intentions.

Reply:
> This should have been said in #221 and it didn't happen because it wasn't a priority yet. If you consider this an issue with high priority you could just say so during a planning and I would happily work on it.

Because I wasn't around when #221 was discussed, I cannot propose changing it now? I think having two functions in the code that do almost the same thing should be merged. I wanted to do it myself, because that was quicker than asking you to do it during planning.

> This sounds again like something we were discussing but didn't agree on yet. In cases like this we should first create an issue with description.

I haven't proposed anything yet. I was giving some background on why I am doing these cleanups now. I think they stand on their own. Please comment on why you think merging them is a bad idea.

Reply:
>  I wanted to do it myself, because that was quicker than asking you to do it during planning.

I think discussing such things during the planning meeting is the correct way to do it.

> Please comment on why you think merging them is a bad idea.

Without further context (the planning), it goes the opposite direction that I had in mind with these changes.